### ISSUE TITLE
No way to track responses which have not been sent to Alaveteli

### ISSUE THREAD

**Issue by lizconlan** [mysociety/foi-register#261]

**Comment by lizconlan** [comment]
something like a `sync_postponed` or `sync_deferred` flag in the db? and/or something in the log?

**Comment by garethrees** [comment]
I think it should come back with a `remote_id` if it has been synced

**Comment by lizconlan** [comment]
a `remote_id` equivalent for responses (note to self: not the same as the `remote_id` for locally created requests) could be a useful addition, ideally this would be the alaveteli incoming message id as that feels like it would be useful for edit tracking, cross-checking, etc

**Comment by garethrees** [comment]
>  ideally this would be the alaveteli incoming message id 

Yeah that's what I was thinking.

> something like "n/a", "unknown", or "not collected"

Just `NULL` – a presenter can then handle what to do if none exist.

**Comment by lizconlan** [comment]
> > something like "n/a", "unknown", or "not collected"
>
> Just NULL – a presenter can then handle what to do if none exist.

The only reason I thought of having a non-default value for historic data is to differentiate between responses that need to be [re]sent and responses that were created - and sent - before the marker was introduced. That said, we already have some "ignore this type of error before a set data" logic in health checks so presumably we could build that logic into any syncing loop as well.

**Comment by garethrees** [comment]
Yeah I'd add a separate marker to historic ones otherwise you'll be converting the `remote_id` to an int all the time and that would be yuck

**Comment by lizconlan** [comment]
It would be a bit odd but we could assign a specific meaning to `0` as this shouldn't ever be a real `remote_id` so `NULL` = not synced, `0` = intentionally left blank, e.g. `123456` = all is well, this a remote response id

**Comment by garethrees** [comment]
That would fail a uniqueness validation, and isn't very rails-way.

I'd add an extra column along the lines of `tracked_remote_message`. Set the default value to `true`, and then write `false` to all existing messages.

**Comment by lizconlan** [comment]
we _could_ have a custom validator that passed 0 or NULL, but you're right we probably shouldn't

**Comment by garethrees** [comment]
> we ___could___ have a custom validator that passed 0 or NULL, but you're right we probably ___shouldn't___

Yeah, it just adding confusion and potential for something to go wrong.
